userland/AppMakefile.mk

```diff
@@ -48,16 +48,50 @@ LIBS += $(TOCK_USERLAND_BASE_DIR)/libc++/libsupc++.a
 LIBS += $(TOCK_USERLAND_BASE_DIR)/libc++/libgcc.a
 
 
+
+# Rules to incorporate external libraries
+define EXTERN_LIB_RULES
+EXTERN_LIB_NAME_$(notdir $(1)) := $(notdir $(1))
+
+# If this library has any additional rules, add them
+-include $(1)/Makefile.app
+
+# If this library has an include directory, add it to search path
+ifneq "$$(wildcard $(1)/include)" ""
+  CPPFLAGS += -I$(1)/include
+endif
+
+# Add arch-specific rules for each library
+$$(foreach arch, $$(TOCK_ARCHS), $$(eval LIBS_$$(arch) += $(1)/build/$$(arch)/$(notdir $(1)).a))
+
+endef
+
+# To see the generated rules, run:
+# $(info $(foreach lib, $(EXTERN_LIBS), $(call EXTERN_LIB_RULES,$(lib))))
+$(foreach lib, $(EXTERN_LIBS), $(eval $(call EXTERN_LIB_RULES,$(lib))))
+
+
+
+ifdef LDFLAGS
```

Comment:
No. It's deliberately right before the rule that would normally use that flag. In the world where people only use that part of the build system somehow, they'll still see this.